src/services/item/plugins/invitation/index.ts

```diff
@@ -49,11 +53,79 @@ const plugin: FastifyPluginAsync = async (fastify) => {
     },
     async ({ member, body, params }) => {
       return db.transaction(async (manager) => {
-        return iS.postManyForItem(member, buildRepositories(manager), params.id, body.invitations);
+        const res = await iS.postManyForItem(
+          member,
+          buildRepositories(manager),
+          params.id,
+          body.invitations,
+        );
+        console.log(res);
+        return res;
       });
     },
   );
 
+  // register upload endpoint separately so the multipart otptions only apply to that specific endpoint
+  fastify.register(async (fastify) => {
+    fastify.register(fastifyMultipart, {
+      limits: {
+        fields: MAX_NON_FILE_FIELDS, // Max number of non-file fields (Default: Infinity).
+        fileSize: MAX_FILE_SIZE, // For multipart forms, the max file size (Default: Infinity).
+        files: MAX_FILES, // Max number of file fields (Default: Infinity).
+      },
+    });
+
+    // post invitations from a csv file
+    fastify.post<{ Params: IdParam; Querystring: { templateId: string } }>(
+      '/:id/invitations/upload-csv',
+      { preHandler: fastify.verifyAuthentication },
+      async (request) => {
+        const { member, query, params } = request;
+        // this is needed to assert the type of the member to be a Member and not an Actor.
+        // verifyAuthentication preHandler should throw if there is no member, but the type can not be narrowed automatically
+        if (!member) {
+          throw new UnauthorizedMember(member);
+        }
+
+        // We need to get the membership service here because it is defined after the invitation service
+        const { memberships } = fastify;
```

Comment:
You can't invert the registering order to prevent this?

Reply:
@pyphilia The `invitation` plugin is registered by the item API but depends on the item API and the `memberships` service
The `memberships` API is registered after the `item` API and depends on `item` and `mailer`. 

I think it might be possible if we instantiate the `memberships` as a plugin inside `item` before the invitations.
Or if we move invitations completely out of `item` and as its own service that depends on item and memberships. 

For now I think it is ok to de-structure it inside the request ? 

We have to map-out the services and their dependencies. Or use a tool like ts-syringe for dependency injection.